Removing json_available reduces backward compatibility

### Expected Behavior

json_available should be available

```python
from flask import json_available
```

### Actual Behavior

```pytb
ImportError: cannot import name 'json_available'
```

### Environment

* Python version: 3.7.3
* Flask version: 1.1.0
* Werkzeug version: 0.15.4

After upgrading to 1.1.0 I'm now getting this same import error with flask-testing at version of 0.7.1.  For now I will lock to the old version of Flask, but it would be nice to get this fixed.  Thanks.
```    from flask_testing import TestCase
/home/*/.local/lib/python2.7/site-packages/flask_testing/__init__.py:13: in <module>
    from .utils import TestCase, LiveServerTestCase
/home/*/.local/lib/python2.7/site-packages/flask_testing/utils.py:38: in <module>
    from flask import json_available, templating, template_rendered
E   ImportError: cannot import name json_available
```

--- a/CHANGES.rst
+++ b/CHANGES.rst
@@ -1,5 +1,15 @@
 .. currentmodule:: flask
 
+Version 1.1.1
+-------------
+
+Unreleased
+
+-   The ``flask.json_available`` flag was added back for compatibility
+    with some extensions. It will raise a deprecation warning when used,
+    and will be removed in version 2.0.0. :issue:`3288`
+
+
 Version 1.1.0
 -------------
 

--- a/src/flask/__init__.py
+++ b/src/flask/__init__.py
@@ -17,6 +17,7 @@ from werkzeug.exceptions import abort
 from werkzeug.utils import redirect
 
 from . import json
+from ._compat import json_available
 from .app import Flask
 from .app import Request
 from .app import Response
@@ -56,4 +57,4 @@ from .signals import template_rendered
 from .templating import render_template
 from .templating import render_template_string
 
-__version__ = "1.1.0"
+__version__ = "1.1.1.dev"

--- a/src/flask/_compat.py
+++ b/src/flask/_compat.py
@@ -113,3 +113,33 @@ except ImportError:
     # https://www.python.org/dev/peps/pep-0519/#backwards-compatibility
     def fspath(path):
         return path.__fspath__() if hasattr(path, "__fspath__") else path
+
+
+class _DeprecatedBool(object):
+    def __init__(self, name, version, value):
+        self.message = "'{}' is deprecated and will be removed in version {}.".format(
+            name, version
+        )
+        self.value = value
+
+    def _warn(self):
+        import warnings
+
+        warnings.warn(self.message, DeprecationWarning, stacklevel=2)
+
+    def __eq__(self, other):
+        self._warn()
+        return other == self.value
+
+    def __ne__(self, other):
+        self._warn()
+        return other != self.value
+
+    def __bool__(self):
+        self._warn()
+        return self.value
+
+    __nonzero__ = __bool__
+
+
+json_available = _DeprecatedBool("flask.json_available", "2.0.0", True)

--- a/tests/test_deprecations.py
+++ b/tests/test_deprecations.py
@@ -1,0 +1,12 @@
+import pytest
+
+from flask import json_available
+
+
+def test_json_available():
+    with pytest.deprecated_call() as rec:
+        assert json_available
+        assert json_available == True  # noqa E712
+        assert json_available != False  # noqa E712
+
+    assert len(rec.list) == 3

